localstack/services/sns/provider.py

```diff
@@ -1414,3 +1425,43 @@ def unsubscribe_sqs_queue(queue_url):
             sub_url = subscriber.get("sqs_queue_url") or subscriber["Endpoint"]
             if queue_url == sub_url:
                 subscriptions.remove(subscriber)
+
+
+def register_sns_api_resource():
```

Comment:
you could use the `@singleton_factory` decorator here that makes sure the method is only called once and in a thread safe way. you can then get rid of the global flag

Reply:
Pretty cool! Used it now, very practical